### ISSUE TITLE
Changing `position` value in `scale_` interacts strangely with `coord_trans()`

### ISSUE THREAD

**Issue by courtiol** [tidyverse/ggplot2#4551]
Applying any transformation using `coord_trans()` modifies what the argument `position` does in `scale_`:
(notice that only the title of the axis is sent to the top and no longer the labels when `coord_trans()` is used)
 
``` r
library(ggplot2)
  
packageVersion("ggplot2")  ## note: current devel does also behave in the same way
#> [1] '3.3.5'

ggplot(data = iris) +
  geom_point(aes(y = Sepal.Length, x = Species)) +
  scale_x_discrete(position = "top")
```

![](https://i.imgur.com/dOEg55w.png)

``` r
ggplot(data = iris) +
  geom_point(aes(y = Sepal.Length, x = Species)) +
  scale_x_discrete(position = "top") +
  coord_trans(y = "identity")
```

![](https://i.imgur.com/UUChPe2.png)

<sup>Created on 2021-07-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

The same goes for the left/right positions.

**Comment by teunbrand** [comment]
This might be related to #3959, but because "Species" is on top it would suggests that facets are aware of the position but coords might not be.

**Comment by thomasp85** [comment]
@paleolimbot all of these issues seems to be related to the fact that only `coord_cartesian()`was moved over to the new axis guide generation pipeline... are you up for moving the remaining coords?

**Comment by yutannihilation** [comment]
This will be fixed by #3972.

``` r
devtools::load_all("~/GitHub/ggplot2/")
#> ℹ Loading ggplot2

ggplot(data = iris) +
  geom_point(aes(y = Sepal.Length, x = Species)) +
  scale_x_discrete(position = "top") +
  coord_trans(y = "identity")
```

![](https://i.imgur.com/yfC96kd.png)

<sup>Created on 2022-07-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
